### PR TITLE
Clarify data collection in Grafana Agent

### PR DESCRIPTION
**Pull request by clayton-cornell**
#### PR Description

This PR moves Data Collection info to a more central location and links from install topics back to data collection.

#### PR Checklist

- [ ] CHANGELOG.md updated
- [ ] Documentation added
- [ ] Tests updated
- [ ] Config converters updated

### CODE REVIEW

File: docs/sources/data-collection.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+The usage information includes the following details:
+
+* A randomly generated, anonymous unique ID (UUID).
+* Timestamp of when the UID was first generated.
+* Timestamp of when the report was created (by default, every four hours).
+* Version of running Grafana Agent.
+* Operating system Grafana Agent is running on.
+* System architecture Grafana Agent is running on.
+* List of enabled feature flags.
+* List of enabled integrations.
```
Comment (rfratto) [inline docs/sources/data-collection.md:30]: These two are only sent for static mode; should we specify that?

File: docs/sources/data-collection.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+---
+
+# Data collection
+
+By default, Grafana Agent sends anonymous but uniquely identifiable usage information from
+your Grafana Agent instance to Grafana Labs. These statistics are sent to `stats.grafana.org`.
+
+Statistics help us better understand how Grafana Agent is used. This helps us prioritize features and documentation.
+
+The usage information includes the following details:
```
Comment (rfratto) [inline docs/sources/data-collection.md:21]: This is missing `List of enabled components` for Flow mode; should it be specified?

**Review by rfratto — APPROVED** [review]
Thanks Clayton! Already an improvement :)

File: docs/sources/data-collection.md (hunk 1 above)
Comment (clayton-cornell) [reply docs/sources/data-collection.md:30]: I'll clarify in a new commit

File: docs/sources/data-collection.md (hunk 2 above)
Comment (clayton-cornell) [reply docs/sources/data-collection.md:21]: Added in a new commit

**Pull request merged**